### PR TITLE
Graph execution

### PR DESCRIPTION
**Pull request by franklinselva**
The idea is to extend the current execution approach to support parallel and asynchronous behavior for the actions produced by the UP. One direction that supports the idea is the use of **dependency graphs** for handling parallelism and reactive behaviors.

This PR implements the entrypoint of using dependency graphs for plan execution. Currently, the execution is straightforward without any condition evaluations and is planned to be implemented later.

### CODE REVIEW

**Comment by alexander-sung** [comment]
Please let us know when the pull request is complete.

**Comment by franklinselva** [comment]
@alexander-sung @sebastianstock 

I think the work is now ready to be merged. Please let me know if there is any change that may be required.

File: up_bridge/plexmo/dispatcher.py [hunk 1]
```diff
@@ -76,3 +77,84 @@ def status(self) -> str:
     def set_replan_callback(self, callback):
         """Set callback function that triggers replanning"""
         self._replan_cb = callback
+
+
+class PlanDispatcher:
```
Comment (sebastianstock) [inline up_bridge/plexmo/dispatcher.py:82]: `PlanDispatcher` and the old `SequentialPlanDispatcher` should be refactored as there is much overlap between them. Currently this is not represented in the code. 
I can do that after this PR is merged.

Comment (sebastianstock) [reply up_bridge/plexmo/dispatcher.py:82]: Probably the `SequentialPlanDispatcher` is obsolete as it serves only a special case of parallel plan execution.

File: up_bridge/components/graph.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    for i, (start, action, duration) in enumerate(plan.timed_actions):
+        child = action.action.name
+        duration = float(duration.numerator) / float(duration.denominator)
+        child_name = f"{child}{action.actual_parameters}({duration}s)"
+        dependency_graph.add_node(child_name, action=child, parameters=action.actual_parameters)
+        dependency_graph.add_edge(parent, child_name, weight=duration)
+        if i + 1 < len(plan.timed_actions):
+            next_start, next_action, next_duration = plan.timed_actions[i + 1]
+            next_duration = float(next_duration.numerator) / float(next_duration.denominator)
+            if start != next_start:
```
Comment (sebastianstock) [inline up_bridge/components/graph.py:65]: Creating the dependency graph from a time triggered plan currently only relies on the actions' start times. I think this assumption is too simplistic for plans with parallel actions. For example, if there are two sequences of actions that should  run in parallel, their dependencies get mixed up.

**Review by sebastianstock — APPROVED** [review]
The main point of this PR, e.g., supporting durative actions and providing first steps towards parallel plan execution looks good. In my opinion we can merge it and  continue working on the execution part from there.

**Review by alexander-sung — APPROVED** [review]

**Pull request merged**